### PR TITLE
StacheAffixTopDirective, StacheAffixComponent

### PR DESCRIPTION
**Pull request by Blackbaud-BrandonJones**
Re-making this Pr to get out of the crazy squirrels nest of merge conflicts now on the other branch

### CODE REVIEW

**Comment by Blackbaud-BrandonJones** [comment]
@Blackbaud-SteveBrush 
- [ ] Sidebar stickiness acts "jumpy" on mobile views. Probably should disable it for mobile, to be honest.
- [ ] I know we don't want to use global styles, but it would be helpful to include classnames to the element depending on its status (e.g., take a look at [these Boostrap styles](http://getbootstrap.com/javascript/#affix)).
- [.] The name of the directive is...lengthy. Something like `stacheAffix` would be nice. 😠 done.
- [ ] You'll need to write unit tests for this.

**Comment by Blackbaud-BrandonJones** [comment]
Brandon Hare: You'll also want to write docs for this. :)

File: src/app/public/src/modules/layout/layout-sidebar.component.scss [hunk 1]
```diff
@@ -0,0 +1,11 @@
+@import '../../styles/_variables';
+
+/deep/
+.affix-top {
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/src/modules/layout/layout-sidebar.component.scss]: `.stache-affix-top`

File: src/app/public/src/modules/shared/affix-top.directive.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { Directive, ElementRef, HostListener, Renderer } from '@angular/core';
+
+import { StacheWindowRef } from './window-ref';
+
+@Directive({
+  selector: '[stacheAffixTop]'
+})
+export class StacheAffixTopDirective {
+    private componentOffset: number;
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/src/modules/shared/affix-top.directive.ts]: One-too-many tabs, here.

File: src/app/public/src/modules/shared/affix-top.directive.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  @HostListener('window:scroll')
+  public onWindowScroll() {
+    if (!this.componentOffset) {
+      this.componentOffset = this.elementRef.nativeElement.offsetTop;
+    }
+
+    let windowOffset = this.windowRef.nativeWindow.pageYOffset;
+
+    if (this.componentOffset <= windowOffset) {
+      this.setElementClass('affix-top', true);
```
Comment (Blackbaud-SteveBrush) [inline src/app/public/src/modules/shared/affix-top.directive.ts]: `.stache-affix-top`

**Comment by Blackbaud-SteveBrush** [comment]
https://github.com/blackbaud/stache2/issues/118

**Pull request merged**